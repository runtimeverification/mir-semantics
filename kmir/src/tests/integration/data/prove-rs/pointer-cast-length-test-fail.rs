../exec-smir/pointers/pointer-cast-length-test-fail.rs