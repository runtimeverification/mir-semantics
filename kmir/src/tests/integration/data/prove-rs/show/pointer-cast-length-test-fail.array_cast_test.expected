
┌─ 1 (root, init)
│   #execTerminator ( terminator ( ... kind: terminatorKindCall ( ... func: operandC
│   span: 0
│
│  (43 steps)
├─ 3 (split)
│   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 2 ) )
┃
┃ (branch)
┣━━┓ subst: .Subst
┃  ┃ constraint:
┃  ┃     notBool size ( ARG_ARRAY1:List ) >=Int 4
┃  │
┃  ├─ 4
┃  │   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 2 ) )
┃  │
┃  │  (4 steps)
┃  └─ 6 (stuck, leaf)
┃      #execTerminator ( terminator ( ... kind: terminatorKindCall ( ... func: operandC
┃      span: 72
┃
┗━━┓ subst: .Subst
   ┃ constraint:
   ┃     size ( ARG_ARRAY1:List ) >=Int 4
   │
   ├─ 5
   │   #selectBlock ( switchTargets ( ... branches: branch ( 0 , basicBlockIdx ( 2 ) )
   │
   │  (208 steps)
   ├─ 7
   │   #traverseProjection ( toStack ( 1 , local ( 2 ) ) , Range ( range ( #mapOffset (
   │   span: 87
   ┃
   ┃ (1 step)
   ┣━━┓
   ┃  │
   ┃  ├─ 8
   ┃  │   #traverseProjection ( toStack ( 1 , local ( 2 ) ) , project:Value ( range ( #map
   ┃  │   span: 87
   ┃  │
   ┃  │  (6 steps)
   ┃  ├─ 10
   ┃  │   #traverseProjection ( toStack ( 1 , local ( 2 ) ) , Range ( #mapOffset ( range (
   ┃  │   span: 87
   ┃  ┃
   ┃  ┃ (1 step)
   ┃  ┣━━┓
   ┃  ┃  │
   ┃  ┃  ├─ 11
   ┃  ┃  │   #traverseProjection ( toStack ( 1 , local ( 2 ) ) , Range ( range ( #mapOffset (
   ┃  ┃  │   span: 87
   ┃  ┃  │
   ┃  ┃  │  (111 steps)
   ┃  ┃  └─ 13 (stuck, leaf)
   ┃  ┃      #traverseProjection ( toLocal ( 5 ) , Range ( range ( #mapOffset ( ARG_ARRAY1:Li
   ┃  ┃      span: 97
   ┃  ┃
   ┃  ┗━━┓
   ┃     │
   ┃     └─ 12 (stuck, leaf)
   ┃         #traverseProjection ( toStack ( 1 , local ( 2 ) ) , Range ( #mapOffset ( range (
   ┃         span: 87
   ┃
   ┗━━┓
      │
      └─ 9 (stuck, leaf)
          #traverseProjection ( toStack ( 1 , local ( 2 ) ) , Range ( range ( #mapOffset (
          span: 87


┌─ 2 (root, leaf, target, terminal)
│   #EndProgram ~> .K


